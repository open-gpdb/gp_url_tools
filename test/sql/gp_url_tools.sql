CREATE EXTENSION url_encode;

SELECT url_encode('Hello World');
SELECT url_decode('Hello%20World');

SELECT url_encode(unnest) from unnest(string_to_array('http://hu.wikipedia.org/wiki/São_Paulo','/'));

SELECT uri_encode('http://hu.wikipedia.org/wiki/São_Paulo');
SELECT md5(uri_decode('http://hu.wikipedia.org/wiki/S%C3%A3o_Paulo'));

SELECT md5(url_decode('%u6D6A%u82B1%u4E00%u6735%u6735%20%u7B2C8%u96C6%20-%20%u89C6%u9891%u5728%u7EBF%u89C2%u770B%20-%20%u6D6A%u82B1%u4E00%u6735%u6735%20-%20%u8292%u679CTV'));
