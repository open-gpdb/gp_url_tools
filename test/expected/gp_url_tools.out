CREATE EXTENSION url_encode;
SELECT url_encode('Hello World');
  url_encode   
---------------
 Hello%20World
(1 row)

SELECT url_decode('Hello%20World');
 url_decode  
-------------
 Hello World
(1 row)

SELECT url_encode(unnest) from unnest(string_to_array('http://hu.wikipedia.org/wiki/São_Paulo','/'));
    url_encode    
------------------
 http%3A
 
 hu.wikipedia.org
 wiki
 S%C3%A3o_Paulo
(5 rows)

SELECT uri_encode('http://hu.wikipedia.org/wiki/São_Paulo');
                 uri_encode                  
---------------------------------------------
 http://hu.wikipedia.org/wiki/S%C3%A3o_Paulo
(1 row)

SELECT md5(uri_decode('http://hu.wikipedia.org/wiki/S%C3%A3o_Paulo'));
               md5                
----------------------------------
 147ded7d471df9cf050bc13242cbf39e
(1 row)

SELECT md5(url_decode('%u6D6A%u82B1%u4E00%u6735%u6735%20%u7B2C8%u96C6%20-%20%u89C6%u9891%u5728%u7EBF%u89C2%u770B%20-%20%u6D6A%u82B1%u4E00%u6735%u6735%20-%20%u8292%u679CTV'));
               md5                
----------------------------------
 d155b1f894fcd5540ba5881fb71753e1
(1 row)

